File: my-folder/1737-maximum-nesting-depth-of-the-parentheses/solution.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> par;
        int maxi = 0;
        for(char c:s){
            if(c=='(') {
                par.push(c);
                if(par.size()>=maxi)maxi = par.size();

            }
            else if(c==')') par.pop();
        }
    return maxi;
    }
};

File: my-folder/1737-maximum-nesting-depth-of-the-parentheses/solution.java
class Solution {
    public int maxDepth(String s) {
        int count = 0;
        int ans = Integer.MIN_VALUE;
        for(char c:s.toCharArray()){
            if(c=='(') count++;
            else if(c==')') count--;
            ans = Math.max(count,ans);
        }
        return ans;
    }
}
